src/store/bond/actions.js

```diff
@@ -0,0 +1,398 @@
+import * as Types from './types'
+import {
+  getBondContract,
+  getOldBondContract,
+  BOND_ADDR,
+} from '../../utils/web3Bond'
+import { getSpartaContract } from '../../utils/web3Sparta'
+import { payloadToDispatch, errorToDispatch } from '../helpers'
+import { getProviderGasPrice } from '../../utils/web3'
+
+export const bondLoading = () => ({
+  type: Types.BOND_LOADING,
+})
+
+// --------------------------------------- BOND+MINT HELPERS ---------------------------------------
+
+export const getBondListed = () => async (dispatch) => {
+  dispatch(bondLoading())
+  const contract = getBondContract()
+
+  try {
+    const bondListed = await contract.callStatic.allListedAssets()
+    dispatch(payloadToDispatch(Types.GET_BOND_LISTED, bondListed))
+  } catch (error) {
+    dispatch(errorToDispatch(Types.BOND_ERROR, error))
+  }
+}
+
+export const getBondListedAsset = (asset) => async (dispatch) => {
+  dispatch(bondLoading())
+  const contract = getBondContract()
+
+  try {
+    const isListed = await contract.callStatic.isListed(asset)
+    dispatch(payloadToDispatch(Types.GET_BOND_LISTED_ASSET, isListed))
+  } catch (error) {
+    dispatch(errorToDispatch(Types.BOND_ERROR, error))
+  }
+}
+
+export const getBondDepositEstimate = (asset, amount) => async (dispatch) => {
+  dispatch(bondLoading())
+  const contract = getBondContract()
+
+  try {
+    const depositEstimate = await contract.callStatic.handleTransferIn(
+      asset,
+      amount,
+    )
+    dispatch(
+      payloadToDispatch(Types.GET_BOND_DEPOSIT_ESTIMATED, depositEstimate),
+    )
+  } catch (error) {
+    dispatch(errorToDispatch(Types.BOND_ERROR, error))
+  }
+}
```

Comment:
handleTransferIn() is not in the ABI sorry, not sure why!
Ill remove it and for the estimate LP output in the UI we can just use either:
- ROUTER.callStatic.addLiquidity()
- UTILS.callstatic.calcTokenPPinBase()